grails-rundeck-data-shared/src/main/groovy/rundeck/data/util/NodeFiltersUtil.groovy

```diff
@@ -0,0 +1,37 @@
+package rundeck.data.util
+
+import com.dtolabs.rundeck.core.utils.NodeSet
+import rundeck.support.filters.BaseNodeFilters
+
+class NodeFiltersUtil {
+    static NodeSet filtersAsNodeSet(BaseNodeFilters econtext) {
+        final NodeSet nodeset = new NodeSet();
+        nodeset.createExclude(BaseNodeFilters.asExcludeMap(econtext)).setDominant(econtext.nodeExcludePrecedence ? true : false);
+        nodeset.createInclude(BaseNodeFilters.asIncludeMap(econtext)).setDominant(!econtext.nodeExcludePrecedence ? true : false);
+        return nodeset
+    }
+
+    static Map extractApiNodeFilterParams(Map params){
+        def result=[:]
+        def value=false
+        //convert api parameters to node filter parameters
+        BaseNodeFilters.filterKeys.each{k,v->
+            if(params[k]){
+                result["nodeInclude${v}"]=params[k]
+                value=true
+            }
+            if(params["exclude-"+k]){
+                result["nodeExclude${v}"]=params["exclude-"+k]
+                value = true
+            }
+        }
+        if(params.filter){
+            result.filter=params.filter
+            value=true
+        }
+        if (value && null!=params.'exclude-precedence') {
+            result.nodeExcludePrecedence = params['exclude-precedence'] == 'true'
```

Comment:
Is `params['exclude-precedence']` always a string?

Reply:
yes